homeassistant/components/deconz/scene.py

```diff
@@ -31,11 +32,14 @@ def async_add_scene(
         | ValuesView[PydeconzScene] = gateway.api.scenes.values(),
     ) -> None:
         """Add scene from deCONZ."""
-        entities = [
-            DeconzScene(scene, gateway)
-            for scene in scenes
-            if scene.deconz_id not in gateway.entities[DOMAIN]
-        ]
+        entities = []
+
+        for scene in scenes:
+
+            known_entities = set(gateway.entities[DOMAIN])
+            new_entity = DeconzScene(scene, gateway)
+            if new_entity.unique_id not in known_entities:
```

Comment:
Is would be a bit more efficient if you could avoid creating the entity object until you know the unique id is not in known entities.